File: unitytogradle/src/main/java/com/commit451/unitytogradle/Structure.java
package com.commit451.unitytogradle;

import java.io.File;

/**
 * The project structure
 */
public class Structure {

    public static Structure create() {
        Structure structure = new Structure();
        structure.project = new File("gradle-project");
        structure.project.mkdir();

        structure.app = new File(structure.project, "app");
        structure.app.mkdir();

        structure.libs = new File(structure.app, "libs");
        structure.libs.mkdir();

        structure.main = new File(structure.app, "src" + File.separator + "main");
        structure.main.mkdirs();

        structure.assets = new File(structure.main, "assets");
        structure.assets.mkdir();

        structure.java = new File(structure.main, "java");
        structure.java.mkdir();

        structure.jniLibs = new File(structure.main, "jniLibs");
        structure.jniLibs.mkdir();

        structure.res = new File(structure.main, "res");
        structure.res.mkdir();
        return structure;
    }

    public File project;
    public File app;
    public File libs;
    public File main;
    public File assets;
    public File java;
    public File jniLibs;
    public File res;

    private Structure() {
    }
}

File: unitytogradle/src/main/java/com/commit451/unitytogradle/Structure.kt
package com.commit451.unitytogradle

import java.io.File

/**
 * The project structure
 */
class Structure private constructor() {

    companion object {

        fun create(): Structure {
            val structure = Structure()
            structure.project = File("gradle-project")
            structure.project.mkdir()

            structure.app = File(structure.project, "app")
            structure.app.mkdir()

            structure.libs = File(structure.app, "libs")
            structure.libs.mkdir()

            structure.main = File(structure.app, "src" + File.separator + "main")
            structure.main.mkdirs()

            structure.assets = File(structure.main, "assets")
            structure.assets.mkdir()

            structure.java = File(structure.main, "java")
            structure.java.mkdir()

            structure.jniLibs = File(structure.main, "jniLibs")
            structure.jniLibs.mkdir()

            structure.res = File(structure.main, "res")
            structure.res.mkdir()
            return structure
        }
    }

    lateinit var project: File
    lateinit var app: File
    lateinit var libs: File
    lateinit var main: File
    lateinit var assets: File
    lateinit var java: File
    lateinit var jniLibs: File
    lateinit var res: File
}
